File: src/main/java/org/micoli/php/symfony/list/DoctrineEntityElementDTO.java
package org.micoli.php.symfony.list;

import com.intellij.psi.PsiElement;
import java.util.List;
import org.jetbrains.annotations.NotNull;

public record DoctrineEntityElementDTO(
        @NotNull String className,
        @NotNull String name,
        @NotNull String schema,
        @NotNull String fqcn,
        @NotNull PsiElement element)
        implements SearchableRecord {

    @Override
    public List<@NotNull String> getSearchString() {
        return List.of(className, name, schema, fqcn);
    }
}

File: src/main/kotlin/org/micoli/php/symfony/list/DoctrineEntityElementDTO.kt
package org.micoli.php.symfony.list

import com.intellij.psi.PsiElement
import kotlinx.collections.immutable.ImmutableList
import kotlinx.collections.immutable.persistentListOf

@JvmRecord
data class DoctrineEntityElementDTO(
    val className: String,
    val name: String,
    val schema: String,
    val fqcn: String,
    val element: PsiElement
) : SearchableRecord {
    override fun getSearchString(): ImmutableList<String> {
        return persistentListOf(className, name, schema, fqcn)
    }
}
